Userland/Utilities/jail-attach.cpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
+#include <unistd.h>
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    int jail_index = 0;
+    Vector<StringView> command;
+    Core::ArgsParser args_parser;
+    bool preserve_env = false;
+    args_parser.set_stop_on_first_non_option(true);
+    args_parser.add_option(preserve_env, "Preserve user environment when running command", "preserve-env", 'E');
+    args_parser.add_positional_argument(jail_index, "Jail Index", "jail index");
+    args_parser.add_positional_argument(command, "Command to execute", "command");
+    args_parser.parse(arguments);
+
+    if (jail_index < 0) {
+        return Error::from_string_view("Can't join jail with negative ID."sv);
+    }
+
+    TRY(Core::System::join_jail(jail_index));
+
+    Vector<StringView> exec_environment;
+    for (size_t i = 0; environ[i]; ++i) {
+        StringView env_view { environ[i], strlen(environ[i]) };
+        auto maybe_needle = env_view.find('=');
+
+        if (!maybe_needle.has_value())
+            continue;
+
+        // FIXME: Allow a custom selection of variables once ArgsParser supports options with optional parameters.
+        if (!preserve_env && env_view.substring_view(0, maybe_needle.value()) != "TERM"sv)
```

Comment:
This looks suspiciously copied from `pls` and `Escalator`. Maybe we should move it into a common LibCore function?